### PR TITLE
Add snapshot tests

### PR DESCRIPTION
**Pull request by MicheleBertoli**
I wrote a few tests to try the new version of `jest-styled-components` and find out some edge cases, feel free to merge this if you think it adds value to the project.

screen shot 2017-07-18 at 15 08 01

### CODE REVIEW

**Comment by philpl** [comment]
This is truly amazing! Thanks a lot, Michele :+1: :tada: :100:

cc @mxstbr, do you want to take a quick look as well, before I merge this?

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "presets": [
+    "env",
```
Comment (philpl) [inline .babelrc]: Just noticed this small change :) It shouldn't be necessary (and might actually cause problems with the module transpilation?) https://github.com/zeit/next.js/blob/v3-beta/server/build/babel/preset.js#L16

Comment (MicheleBertoli) [inline .babelrc]: Unfortunately it's necessary for testing, and removing it makes Jest complain:
> SyntaxError: Unexpected token import

I updated the `.babelrc` following [the example](https://github.com/zeit/next.js/blob/v3-beta/examples/with-jest/.babelrc) on the repo of Next.js, so there's should be no problems.

Comment (philpl) [inline .babelrc]: Perfect! :)

**Comment by philpl** [comment]
I'm going to go ahead an merge this and will set up Travis later :)

**Comment by MicheleBertoli** [comment]
🙌 Yay, thanks @philpl.
